### ISSUE TITLE
Methods not recognized for implemented generic trait

### ISSUE THREAD

**Issue by johnthagen** [intellij-rust/intellij-rust#743]
`Cargo.toml`:
```toml
[package]
name = "rust_test"
version = "0.1.0"
authors = ["johnthagen <johnthagen@gmail.com>"]

[dependencies]
piston_window = "0.54.0"
```
`main.rs`:
```rust

extern crate piston_window;

use piston_window::{EventLoop, PistonWindow, WindowSettings};

fn main() {
    let mut window: PistonWindow = WindowSettings::new("Window", [1024, 600])
        .exit_on_esc(true).build().unwrap();

    window.set_ups(60);  // <--- Not recognized as method
    window.set_max_fps(60);  // <--- Not recognized as method
}
```

Documentation for methods:
- [`set_ups`](http://docs.piston.rs/piston_window/event_loop/trait.EventLoop.html#tymethod.set_ups)
- [`set_max_fps`](http://docs.piston.rs/piston_window/event_loop/trait.EventLoop.html#tymethod.set_max_fps)
- [Implementation of `EventLoop` trait](http://docs.piston.rs/piston_window/src/piston_window/src/lib.rs.html#320-338)

**Comment by alobb** [comment]
I'm trying to recreate this locally, but can't figure out what is wrong.  Can you clarify what behavior is missing that is expected (eg., syntax highlighting works for me, and cmd-click on `set_ups` and `set_max_fps` bring me to the correct function declarations on lines 323 and 327 of `lib.rs` in the `piston_window` library.  I'm running master (at the time of this writing, commit hash `bd2589a8e68e796a5104f30e2245de5fc63b6d62`), can you still reproduce?

**Comment by johnthagen** [comment]
Confirmed that this has been fixed on `intellij-rust` 0.1.0.1129.

Tested in:
- PyCharm 2016.2.3
- Rust 1.12.1
- Windows 10 x64
